pkg/lib/indexer/indexer.go

```diff
@@ -513,38 +523,72 @@ func (i ImageIndexer) ExportFromIndex(request ExportFromIndexRequest) error {
 	}
 
 	var errs []error
+	var g errgroup.Group
+	var mu = &sync.Mutex{}
+
+	concurrentGoroutines := make(chan struct{}, concurrencyLimitForExport)
+
 	for _, bundleImage := range bundles {
-		// try to name the folder
-		folderName, err := dbQuerier.GetBundleVersion(context.TODO(), bundleImage)
-		if err != nil {
+		bundleImage := bundleImage // to ensure every go routine gets a it's own copy of closure variable
+
+		g.Go(func() error {
+			concurrentGoroutines <- struct{}{}
+			defer func() { <-concurrentGoroutines }()
+
+			// get the bundle version to be used in naming the folder
+			bundleVersion, err := dbQuerier.GetBundleVersion(context.TODO(), bundleImage)
+			if err != nil {
+				i.Logger.Errorf("error getting bundle version from image: %s", err)
+				return err
+			}
+			if bundleVersion == "" {
+				// operator-registry does not care about the folder name
+				bundleVersion = bundleImage
+			}
+			// fetch bundle name from bundle image
+			pkgName := strings.FieldsFunc(bundleImage, splitBundlePath)[2]
```

Comment:
String manipulation like this has a tendency to be brittle. Could you use something from https://godoc.org/github.com/docker/distribution/reference?

Reply:
modified code doesn't use this anymore